functions/Remove-DbaDbData.ps1

```diff
@@ -0,0 +1,166 @@
+                    Write-Message -Level Verbose -Message "Processing DbaInstanceParameter through InputObject"
+                    $dbDatabases = Get-DbaDatabase -SqlInstance $input -SqlCredential $sqlcredential -Database $Database -ExcludeDatabase $ExcludeDatabase -ExcludeSystem
+                }
+                'Microsoft.SqlServer.Management.Smo.Server' {
+                    Write-Message -Level Verbose -Message "Processing Server through InputObject"
+                    $dbDatabases = Get-DbaDatabase -SqlInstance $input -SqlCredential $sqlcredential -Database $Database -ExcludeDatabase $ExcludeDatabase -ExcludeSystem
+                }
+                'Microsoft.SqlServer.Management.Smo.Database' {
+                    Write-Message -Level Verbose -Message "Processing Database through InputObject"
+                    $dbDatabases = $input | Where-Object { -not $_.IsSystemObject }
+                }
+                default {
+                    Stop-Function -Message "InputObject is not a server or database."
+                    return
+                }
+            }
+
+            foreach ($db in $dbDatabases) {
+                if ($Pscmdlet.ShouldProcess($db.Name, "Removing all data on $($db.Parent.Name)")) {
+                    Write-Message -Level Verbose -Message "Truncating tables in $db on instance $instance"
+                    $instance = $db.Parent
+                    try {
+
+                        # Collect up the objects we need to drop and recreate
+                        $objects = @()
+                        $objects += Get-DbaDbForeignKey -SqlInstance $instance -Database $db.Name
+                        $objects += Get-DbaDbView -SqlInstance $instance -Database $db.Name -ExcludeSystemView
+
+                        # Script out the create statements for objects
+                        $createOptions = New-DbaScriptingOption
+                        $createOptions.Permissions = $true
+                        $createOptions.ScriptBatchTerminator = $true
+                        $createOptions.AnsiFile = $true
+                        $null = $objects | Export-DbaScript -FilePath "$Path\$($db.Name)_Create.Sql" -ScriptingOptionsObject $createOptions
+
+                        # Script out the drop statements for objects
+                        $options = New-DbaScriptingOption
+                        $options.ScriptDrops = $true
+                        $null = $objects | Export-DbaScript -FilePath "$Path\$($db.Name)_Drop.Sql" -ScriptingOptionsObject $options
+                    } catch {
+                        Stop-Function -Message "Issue scripting out the drop\create scripts for objects in $db on instance $instance" -ErrorRecord $_
+                        return
+                    }
+
+                    try {
+                        if ($objects) {
+                            Invoke-DbaQuery -SqlInstance $instance -Database $db.Name -File "$Path\$($db.Name)_Drop.Sql"
```

Comment:
For efficiency, I would recommend using `$server.Query()` instead of `Invoke-DbaQuery`.

Since you are not using `PassThru` on the `Export-DbaScript` call your `$objects` should contain the full path and file name object from `Get-ChildItem` (I recall).

Reply:
Just gives a different way of doing it on the `$objects` processing...

Reply:
When using `$server.query()` would I need to read in the script using Get-Content or something?
Also $objects at this point contains the details of the FKs and Views - am I missing what you mean?
![image](https://user-images.githubusercontent.com/981370/96568069-70798e00-12bf-11eb-8fd4-5f59e1a7ea44.png)